kimchi/src/circuits/constraints.rs

```diff
@@ -403,7 +406,7 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F> {
         fr_sponge_params: ArithmeticSpongeParams<F>,
         public: usize,
         precomputations: Option<Arc<DomainConstantEvaluations<F>>>,
-    ) -> Option<Self> {
+    ) -> Result<Option<Self>, ProverError> {
```

Comment:
why not `Result<Self, ProverError>`? Is there a usecase for `Ok(None)` here?

Reply:
true that, seems like the Option is not needed at all in that case